src/VisualStudio/LiveShare/Impl/RunCodeActionsHandler.cs

```diff
@@ -7,127 +7,74 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.CodeRefactorings;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.Internal.Log;
 using Microsoft.CodeAnalysis.LanguageServer;
 using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
 using Microsoft.CodeAnalysis.LanguageServer.Handler;
-using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.Commands;
 using Newtonsoft.Json.Linq;
 using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
 
 namespace Microsoft.VisualStudio.LanguageServices.LiveShare
 {
     /// <summary>
-    /// Run code actions handler.  Called when lightbulb invoked.
-    /// Code actions must be applied from the UI thread in VS.
+    /// Runs code actions as a command on the server.
+    /// Commands must be applied from the UI thread in VS.
     /// </summary>
-    internal abstract class RunCodeActionsHandler : ILspRequestHandler<LSP.ExecuteCommandParams, object, Solution>
+    [ExportExecuteWorkspaceCommand(CodeActionsHandler.RunCodeActionCommandName)]
+    internal class RunCodeActionsHandler : IExecuteWorkspaceCommandHandler
```

Comment:
This should be moved from the liveshare layer down to where the rest of the handlers are